Lib/concurrent/futures/_base.py

```diff
@@ -191,16 +205,18 @@ def as_completed(fs, timeout=None):
     if timeout is not None:
         end_time = timeout + time.time()
 
+    total_futures = len(fs)
+
     fs = set(fs)
```

Comment:
This is a regression. The ordering of instructions here is wrong. Now `fs` must be a sequence to support the `len(fs)`.